include/jemalloc/internal/hpa_opts.h

```diff
@@ -7,8 +7,53 @@
 /*
  * This file is morally part of hpa.h, but is split out for header-ordering
  * reasons.
+ *
+ * All of these hpa_shard_opts below are experimental. We are exploring more
+ * efficient packing, hugifying, and purging approaches to make efficient
+ * trade-offs between CPU, memory, latency, and usability. This means all of
+ * them are at the risk of being deprecated and corresponding configurations
+ * should be updated once the final version settles.
  */
 
+/*
+ * This enum controls when jemalloc hugifies/dehugifies pages.  Each mode may be
+ * more suitable depending on deployment environments.
+ *
+ * hpa_hugify_style_none
+ * This style only makes sense when deploying on systems where THP are enabled
+ * in 'always' mode.  With this style, you most likely want to have no purging
+ * at all (dirty_mult=-1) or purge_threshold=HUGEPAGE (2097152 for 2Mb page),
+ * although other thresholds may work well depending on kernel settings of your
+ * deployment targets.
+ *
+ * hpa_hugify_style_eager
+ * This style will set 'hg' vm_flag immediately after memory is mapped, and in
+ * case when THP is enabled in 'madvise' mode give you ability to start with
+ * page as huge.  Same options for purging as for the style 'none' are good
+ * starting choices: no purging, or purge_threshold=HUGEPAGE, some
+ * min_purge_delay_ms that allows for page not to be purged quickly, etc.
+ *
+ * hpa_hugify_style_lazy
+ * This style is suitable when you purge more aggressively (you sacrifice CPU
+ * performance for less memory).  If purging with purge_threshold < HUGEPAGE
+ * you probably want this style or 'eager_and_lazy'.
+ *
+ * hpa_hugify_style_eager_and_lazy
+ * As the name suggests, this combines 'eager' and 'lazy' and is suitable when
+ * you want CPU perf form the begining, but want your memory to be tighter(more
+ * aggressive dirty_mult, purge_threshold, higher hugification_threshold, etc.
```

Comment:
Apropos my comments above, these two cases would also benefit from an explanation of what they do.  Since you mention latency in the comment above this one, I think these items should probably mention the latency effects of the more frequent acquisition of kernel locks on Linux.